libvncclient/rfbproto.c

```diff
@@ -2332,12 +2494,19 @@ HandleRFBServerMessage(rfbClient* client)
   case rfbServerCutText:
   {
     char *buffer;
+    int32_t ilen; /* also as a flag, if ilen < 0, it is ext clipboard text */
 
     if (!ReadFromRFBServer(client, ((char *)&msg) + 1,
 			   sz_rfbServerCutTextMsg - 1))
       return FALSE;
 
-    msg.sct.length = rfbClientSwap32IfLE(msg.sct.length);
+    ilen = rfbClientSwap32IfLE(msg.sct.length);
+    if (client->appData.clipboardCap && ilen >= 0) {
```

Comment:
LibVNCClient should be able to deal automatically with both kinds of servers. So rather ilen >= 0 -> use traditional cuttext handling; ilen < 0 -> use extended clipboard handling.

Reply:
it need the vncserver also cope with it too. and it need the lib user to cope with it. the use set utf8 clipboard, then it  will use the utf-8 api from/to os. i  think this is not the  library's duty. let us push  it to upper layer.  In most case, user unset it. If we know for example, tigervncserver, set it.

Reply:
Sorry, have to re-open this one here:

What about the case of a modern client app (with `client->appData.clipboardCap` set to TRUE) and a legacy server that does only support Latin-1 cuttext? I think the client would bail out unnecessarily altough it'd be able to handle Latin-1 cuttext...

This probably needs a check with `SupportsServer2Client()`  (although then it should have been set by `SetServer2Client()`, but it's unclear to me where this could happen...

Or you could just remove this check here.

Reply:
Just log it and let it go....

Reply:
add clipboardEnabledCap to make logic more whole (it is still far from full support).
client use rfb_client->appData.clipboardCap = rfbExtendedClipboard_Text to set the text cap, and use rfb_client->appData.clipboardEnabledCap & rfbExtendedClipboard_Text to check if it is supported by server.